### DISCUSSION TITLE
Planet Scale support & foreign keys

### DISCUSSION THREAD

**Discussion by tvd0x2a** [craftcms/cms#12112] [Ideas] [answered]
Hello, I tried using planetscale.com to host the mysql database for my craft site. It quickly failed, because PlanetScale does not support foreign keys. They have a couple docs explaining why this is—it's mainly because PlanetScale provides online schema migrations and foreign keys make that difficult to do generally.

* https://planetscale.com/docs/learn/operating-without-foreign-key-constraints
* https://vitess.io/blog/2021-06-15-online-ddl-why-no-fk/

Is there a future where Craft CMS works without foreign keys?

**Comment by angrybrad** [comment, marked as answer]
> Is there a future where Craft CMS works without foreign keys?

We've discussed this internally before, and there certainly exists a future in which this could happen.

It would be quite a breaking change, though. And we'd need to do some pretty extensive performance testing.

i.e. if someone deletes a site with 1 million elements, Craft/PHP can rely on cascading deletes of foreign keys now to do the bulk of the work.

If we start enforcing referential integrity at the application level, those cascading deletes would all have to happen from PHP, which would take ages longs.
